### PR TITLE
[quest] ADDED: 'Quick Draw' Fake Rune Quest To QuestDB

### LINKED ISSUE #5431: [quest] 'Quick Draw' Rogue Rune Fake Quest Missing From QuestDB
## Description

The SoD Rogue fake rune quest 'Quick Draw' is not currently in the QuestDB. 

## Wowhead link

https://www.wowhead.com/classic/item=208220/dun-morogh-treasure-map
https://www.wowhead.com/classic/item=203750/elwynn-treasure-map
https://www.wowhead.com/classic/item=208605/teldrassil-treasure-map
https://www.wowhead.com/classic/item=207110/durotar-treasure-map
https://www.wowhead.com/classic/item=208034/tirisfal-treasure-map

## Screenshots

N/a

## Questie & WoW version

v9.2.13
WoW Classic SoD - v1.15.052409

### PR DESCRIPTION
**Pull request by Sommos**
## Issue references

Fixes #5431

## Proposed changes

- ADDED: 'Quick Draw' Fake Rune Quest is now in the QuestDB, and should be accessible to users.

### CODE REVIEW

**Comment by Sommos** [comment]
The required races is set to none as in theory a Rogue can go to any of these places, kill the NPCs and grab the map pieces. 

We can change this if we want to reduce the number of icons being shown to the user.

File: Database/Corrections/sodQuestFixes.lua [hunk 1]
```diff
@@ -2286,6 +2286,66 @@ function SeasonOfDiscovery:LoadQuests()
             [questKeys.requiredSpell] = -410023,
             [questKeys.zoneOrSort] = sortKeys.DRUID,
         },
+        [90133] = {
+            [questKeys.name] = "Quick Draw",
+            [questKeys.startedBy] = {nil,{208220}},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:2291]: The second entry of the `startedBy` table is a list object IDs, not item IDs, as you can see here: https://github.com/Questie/Questie/blob/6f0f1d61d4593447cbb40be01442141b4f4ef2fb/Database/questDB.lua#L5-L47

Even though **in theory** the "quest" is started by the items, it's still easier to simply add the NPCs which drop the map pieces directly. "item" quests are not even added to the map atm, only if we manually add a `creatureStart`.

File: Database/Corrections/sodQuestFixes.lua [hunk 2]
```diff
@@ -2286,6 +2286,66 @@ function SeasonOfDiscovery:LoadQuests()
...
+            [questKeys.finishedBy] = nil,
+            [questKeys.requiredLevel] = 1,
+            [questKeys.questLevel] = 5,
+            [questKeys.requiredRaces] = raceIDs.NONE,
+            [questKeys.requiredClasses] = classIDs.ROGUE,
+            [questKeys.objectivesText] = {"Follow the location of the treasure map and loot the treasure chest."},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:2297]: This should note that you pick pocket the NPCs to retrieve the map pieces.

**Review by BreakBB — CHANGES_REQUESTED** [review]
Besides the things I noted, we should hide this quest till #5362 is done. The map pieces drop from maaaany mobs, so the map would be flooded with icons.

File: Database/Corrections/sodQuestFixes.lua (hunk 2 above)
Comment (Sommos) [reply Database/Corrections/sodQuestFixes.lua]: Fixed in: https://github.com/Questie/Questie/pull/5432/commits/4b6b39a0225ae357fa3b23cdf884a5bcf21336d4

File: Database/Corrections/sodQuestFixes.lua (hunk 1 above)
Comment (Sommos) [reply Database/Corrections/sodQuestFixes.lua]: Fixed in: https://github.com/Questie/Questie/pull/5432/commits/4b6b39a0225ae357fa3b23cdf884a5bcf21336d4

Yes, we will need to wait until #5362 is finished before merging this.

Comment (BreakBB) [reply Database/Corrections/sodQuestFixes.lua]: We can still merge it, but blacklist it for now, like we did with similar runes. There is a table in `SeasonOfDiscoveey.lua` for it.

Comment (BreakBB) [reply Database/Corrections/sodQuestFixes.lua]: Also your commit didn't fix the initial issue.

You wrote item IDs in an object ID field. Not you are writing NPC IDs in an object field.

**Comment by BreakBB** [comment]
It is still required to add the new fake quests to the `runeQuestsInSoD` table. Otherwise they won't be filtered correctly and will show up with a `!` like regular quests.

You should really consider using a level 1 character to see if the runes show up properly.

**Comment by Sommos** [comment]
> It is still required to add the new fake quests to the `runeQuestsInSoD` table. Otherwise they won't be filtered correctly and will show up with a `!` like regular quests.
> 
> You should really consider using a level 1 character to see if the runes show up properly.

Yeah. It's fixed now. I will be doing that going forward to avoid silly mistakes like this.

File: Database/Corrections/sodQuestFixes.lua [hunk 3]
```diff
@@ -2286,6 +2286,66 @@ function SeasonOfDiscovery:LoadQuests()
...
+        },
+        [90135] = {
+            [questKeys.name] = "Quick Draw",
+            [questKeys.startedBy] = {{2022,2027,2029,2030,2025,2010,2152,2009,2011,2013,2012,2019,2021,2015,2020,2018,2017,2005,2002,2004,2003}},
+            [questKeys.finishedBy] = nil,
+            [questKeys.requiredLevel] = 1,
+            [questKeys.questLevel] = 5,
+            [questKeys.requiredRaces] = raceIDs.NONE,
+            [questKeys.requiredClasses] = classIDs.ROGUE,
+            [questKeys.objectivesText] = {"Pickpocket Timberlings, Gnarlpine, Harpies and Sprites for map pieces. Collect all 4, and combine them.Follow the location of the treasure map and loot the treasure chest."},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:2321]:
```suggestion
            [questKeys.objectivesText] = {"Pickpocket Timberlings, Gnarlpine, Harpies and Sprites for map pieces. Collect all 4, and combine them. Follow the location of the treasure map and loot the treasure chest."},
```

File: Database/Corrections/sodQuestFixes.lua [hunk 4]
```diff
@@ -2286,6 +2286,66 @@ function SeasonOfDiscovery:LoadQuests()
...
+        },
+        [90136] = {
+            [questKeys.name] = "Quick Draw",
+            [questKeys.startedBy] = {{3129,2128,3196,3195,3199,3197,3198,3111,3114,3113,3112,3206,3207}},
+            [questKeys.finishedBy] = nil,
+            [questKeys.requiredLevel] = 1,
+            [questKeys.questLevel] = 5,
+            [questKeys.requiredRaces] = raceIDs.NONE,
+            [questKeys.requiredClasses] = classIDs.ROGUE,
+            [questKeys.objectivesText] = {"Pickpocket Kul Tiras, Burning Blade, Quilboar and Trolls for map pieces. Collect all 4, and combine them.Follow the location of the treasure map and loot the treasure chest."},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:2333]:
```suggestion
            [questKeys.objectivesText] = {"Pickpocket Kul Tiras, Burning Blade, Quilboar and Trolls for map pieces. Collect all 4, and combine them. Follow the location of the treasure map and loot the treasure chest."},
```

File: Database/Corrections/sodQuestFixes.lua [hunk 5]
```diff
@@ -2286,6 +2286,66 @@ function SeasonOfDiscovery:LoadQuests()
...
+        },
+        [90137] = {
+            [questKeys.name] = "Quick Draw",
+            [questKeys.startedBy] = {{1535,1536,1538,1540,1537,1539,1935,1934,1545,1543,1544,1941,1675,1674}},
+            [questKeys.finishedBy] = nil,
+            [questKeys.requiredLevel] = 1,
+            [questKeys.questLevel] = 5,
+            [questKeys.requiredRaces] = raceIDs.NONE,
+            [questKeys.requiredClasses] = classIDs.ROGUE,
+            [questKeys.objectivesText] = {"Pickpocket Scarlet, Tirisfal Farmers, Murlocs and Gnolls for map pieces. Collect all 4, and combine them.Follow the location of the treasure map and loot the treasure chest."},
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:2345]:
```suggestion
            [questKeys.objectivesText] = {"Pickpocket Scarlet, Tirisfal Farmers, Murlocs and Gnolls for map pieces. Collect all 4, and combine them. Follow the location of the treasure map and loot the treasure chest."},
```

**Review by BreakBB — APPROVED** [review]

**Pull request merged**